### PR TITLE
Fix `get_method_list` of GDScript to properly fill the returning array

### LINKED ISSUE #33624: get_method_list() returns an incorrect info about custom functions
**Godot version:**
3.1.1.stable.official
3.2.beta1.official

**OS/device including version:**
Linux

**Issue description:**
See the title above and the example below.

**Steps to reproduce:**

```gdscript
extends Node

func my_func(s: String, my_arg := 42000) -> bool:
    return my_arg % 7 == 0

func print_info(i):
    var args = ''
    for j in i.args:
        args += '%s:%s;' % [j.name, j.type]
    print('name=%s args=%s default_args=%s return.type=%s' %
            [i.name, args, i.default_args, i.return.type])

func _ready():
    for i in get_method_list():
        if i.name == 'my_func' || i.name == 'connect':
            print_info(i)
    for i in get_script().get_script_method_list():
        if i.name == 'my_func':
            print_info(i)

# name=connect args=signal:4;target:17;method:4;binds:19;flags:2;
    # default_args=[[], 0] return.type=2
# name=my_func args=arg0:0;arg1:0; default_args=[] return.type=0
# name=my_func args=:4;:2; default_args=[] return.type=1
```

**Minimal reproduction project:**

### PR DESCRIPTION
**Pull request by Chaosus**
Fix https://github.com/godotengine/godot/issues/33624

### CODE REVIEW

**Comment by ryehollister** [comment]
Any updates on this? jw

**Comment by CrezyDud** [comment]
we need this

**Comment by dalexeev** [comment]
I'm not sure, but probably `GDScript::get_method_info()` should also be fixed.

File: modules/gdscript/gdscript.cpp [hunk 1]
```diff
@@ -1482,8 +1490,21 @@ void GDScriptInstance::get_method_list(List<MethodInfo> *p_list) const {
 			mi.name = E.key;
 			mi.flags |= METHOD_FLAG_FROM_SCRIPT;
 			for (int i = 0; i < E.value->get_argument_count(); i++) {
-				mi.arguments.push_back(PropertyInfo(Variant::NIL, "arg" + itos(i)));
```
Comment (dalexeev) [inline modules/gdscript/gdscript.cpp:1485]: It makes sense to leave it in `#else`.

**Comment by adamscott** [comment]
@dalexeev What's your thoughts on this PR to merge it in 4.1? Is it safe/stable enough to merge this before the release freeze (in 10 days already, will only merge critical bug fixes)?

@Chaosus Can you add some unit tests using `get_method_list` with the properly filled returning array?

**Comment by dalexeev** [comment]
It's been 2 years since the last update, this PR is worth rebasing and testing. Adding the missing info shouldn't break anything, but I think we should make sure it's actually correct now.

I agree with you about unit tests. We should also document that in release builds, info may differ from debug builds.

I can do it if @Chaosus doesn't have the ability or desire to do it now for some reason. But do we want to include it urgently in 4.1 or we can postpone it to 4.1.x?

**Comment by adamscott** [comment]
> But do we want to include it urgently in 4.1 or we can postpone it to 4.1.x?

There is no urgency, but it could help people like @MikeSchulze if we're able to merge it. It's a bug after all. If it was only an enhancement PR, it would have been already postponed to 4.2.

Let's give ourselves some time (like until Monday) to decide.

**Comment by akien-mga** [comment]
Superseded by #81079.

**Comment by dalexeev** [comment]
Thanks for the PR nonetheless!

**Pull request closed without merging**